activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -209,10 +209,16 @@ def in_order_of(key, series, filter: true)
   #   Set.new.sole        # => Enumerable::SoleItemExpectedError: no item found
   #   { a: 1, b: 2 }.sole # => Enumerable::SoleItemExpectedError: multiple items found
   def sole
-    case count
-    when 1   then return first # rubocop:disable Style/RedundantReturn
+    if is_a?(Range) && (self.begin.nil? || self.end.nil?)
```

Comment:
Should we redefine `sole` on `Range` then so other `Enumerable` don't have to pay the price of checking `is_a(Range)` on each call? Range's implementation can perform the check and then call `super`

Reply:
Good idea 👍